core/src/main/java/com/mapzen/android/graphics/model/MarkerManager.java

```diff
@@ -27,24 +45,61 @@ public MarkerManager(MapController mapController) {
    */
   public BitmapMarker addMarker(MarkerOptions markerOptions) {
     final Marker marker = mapController.addMarker();
-    marker.setPoint(markerOptions.getPosition());
-    if (markerOptions.getIconDrawable() != null) {
-      marker.setDrawable(markerOptions.getIconDrawable());
-    } else {
-      marker.setDrawable(markerOptions.getIcon());
-    }
-    StyleStringGenerator styleStringGenerator = new StyleStringGenerator();
-    styleStringGenerator.setSize(markerOptions.getWidth(), markerOptions.getHeight());
-    marker.setStylingFromString(styleStringGenerator.getStyleString());
-    return new BitmapMarker(this, marker, styleStringGenerator);
+    BitmapMarker bitmapMarker = bitmapMarkerFactory.createMarker(this, marker,
+        styleStringGenerator);
+    configureMarker(bitmapMarker, markerOptions.getPosition(), markerOptions.getIconDrawable(),
```

Comment:
Instead of passing every property, can we not pass the markerOptions explicitly?

Reply:
I use this method in two places- here when adding markers given `MarkerOptions` and then when restoring markers when I don't have options but instead only the `BitmapMarker` which is why I don't have it taking the `MarkerOptions` explicitly. I'll create a couple convenience methods on top of this one to handle configuring a marker given just the `MarkerOptions` and have it call through to this one. How does that sound? I could keep a mapping of `MarkerOptions` -> `BitmapMarker` and then have just one method to configure a marker but I'd prefer not to store this info if I can get away with it.

Reply:
Yup that sounds fine to me.